portal-impl/src/com/liferay/portlet/ActionResponseFactory.java

```diff
@@ -14,27 +14,44 @@
 
 package com.liferay.portlet;
 
+import aQute.bnd.annotation.ProviderType;
+
 import com.liferay.portal.kernel.model.Layout;
 import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.portlet.LiferayActionResponse;
 
+import javax.portlet.ActionRequest;
 import javax.portlet.PortletException;
+import javax.portlet.filter.ActionRequestWrapper;
 
 import javax.servlet.http.HttpServletResponse;
 
 /**
  * @author Brian Wing Shun Chan
+ * @author Neil Griffin
  */
+@ProviderType
 public class ActionResponseFactory {
 
-	public static ActionResponseImpl create(
-			ActionRequestImpl actionRequestImpl, HttpServletResponse response,
+	public static LiferayActionResponse create(
+			ActionRequest actionRequest, HttpServletResponse response,
 			User user, Layout layout)
 		throws PortletException {
 
+		while (true) {
```

Comment:
@shuyangzhou I just noticed that I didn't correct this `while (true) {` thing. I will do that in a subsequent PR for ActionResponseFactory, EventResponseFactory, HeaderResponseFactory, RenderResponseFactory, and ResourceResponseFactory so don't worry about that.